### DISCUSSION TITLE
Modal custom content script only executes once

### DISCUSSION THREAD

**Discussion by khairulashraff** [filamentphp/filament#4871] [Help]
I am building a custom modal opened through a table action and getting stuck with reloading JS that gets executed on it. Everything renders fine, but the JS only executes once, not every modal open.
```blade
<div>
    <div wire:ignore id="mapContainer" style="width:100%; height:750px;"></div>
</div>

<script>
    const coordinates = {{ Illuminate\Support\Js::from($trip->pings()->map(function($ping) {
        return ['lat' => $ping->location->getLat(), 'lng' => $ping->location->getLng()];
    })) }};

    // map interaction
    ZMap.renderPolyline('mapContainer', coordinates);
</script>

```
The coordinates do change on each modal, but the script is only executed once on the first load/render so the map does not update.

I understand the modal only gets rendered once, the first time around but the docs does not say how to handle JS if the custom content has JS codes as well.

What's the proper way of handling modal custom content with JS executions?

**Comment by Cayaf89** [comment]
For those wondering on the same question, here's a solution using x-init attribute :

```
<div 
    x-init="() => {
        const coordinates = {{ Illuminate\Support\Js::from($trip->pings()->map(function($ping) {
            return ['lat' => $ping->location->getLat(), 'lng' => $ping->location->getLng()];
        })) }};

        // map interaction
        ZMap.renderPolyline('mapContainer', coordinates);
    }"
>
    <div wire:ignore id="mapContainer" style="width:100%; height:750px;"></div>
</div>
```
